### PR TITLE
Implement example batches with modified devon-batch (see #190 in devo…

### PR DESCRIPTION
**Pull request by maihacke**
First version of reworked spring-batch integration (https://github.com/devonfw/devon4j/issues/190).
It requires changes in devon4j-batch (see https://github.com/devonfw/devon4j/pull/191)

### CODE REVIEW

**Comment by vapadwal** [comment]
I tried to run the job by below command as mentioned in the code, is this the only way to do it.

java -jar mtsj-batch-3.2.0-SNAPSHOT-bootified.jar --spring.main.web-application-type=none --spring.batch.job.enabled=true --spring.batch.job.names=ingredientExportJob pathToFile=D:\projects\example.csv


java -jar mtsj-batch-3.2.0-SNAPSHOT-bootified.jar --spring.main.web-application-type=none --spring.batch.job.enabled=true --spring.batch.job.names=ingredientImportJob pathToFile=D:\projects\example.csv

**Comment by maihacke** [comment]
Yes that's the idea.

**Comment by vapadwal** [comment]
this PR can be merge once devon4j 3.3.0 for batch is released , and we need to change the version in mtsj for devon4j to 3.3.0

**Comment by vapadwal** [comment]
when I run the batch as mentioned in my above comments (running job ingredientImportJob ).
I want to see the data that has been updated in INGREDIENT table .

Basically everything runs successfully but the server closes when we run the batch. So wont able to see if the data is updated.

I tried by only running java -jar mtsj-batch-3.2.0-SNAPSHOT-bootified.jar . the server started but again the there was exception thrown. I have checked that it has created the required spring batch tables in db like BATCH_JOB_EXECUTION,BATCH_STEP_EXECUTION_CONTEXT  etc.

basically i want to know is it possible to showcase we can add data to existing mtsj application using spring batch

**Comment by maihacke** [comment]
> when I run the batch as mentioned in my above comments (running job ingredientImportJob ).
> I want to see the data that has been updated in INGREDIENT table .
> 
> Basically everything runs successfully but the server closes when we run the batch. So wont able to see if the data is updated.
> 
> I tried by only running java -jar mtsj-batch-3.2.0-SNAPSHOT-bootified.jar . the server started but again the there was exception thrown. I have checked that it has created the required spring batch tables in db like BATCH_JOB_EXECUTION,BATCH_STEP_EXECUTION_CONTEXT etc.
> 
> basically i want to know is it possible to showcase we can add data to existing mtsj application using spring batch

The easiest way is to use H2 AUTO_SERVER-Mode. You can activate it by changing the datasource URL:

java -jar mtsj-batch-3.2.0-SNAPSHOT-bootified.jar --spring.datasource.url='jdbc:h2:~/mtsbatch;AUTO_SERVER=TRUE' --spring.main.web-application-type=none --spring.batch.job.enabled=true --spring.batch.job.names=ingredientImportJob pathToFile=D:\projects\example.csv

**Comment by hohwille** [comment]
> [ERROR] Non-resolvable import POM: Could not find artifact com.devonfw.java.boms:devon4j-bom:pom:3.3.0-SNAPSHOT @ line 50, column 19

Travis network issue? I restarted the build.

**Comment by maihacke** [comment]
> for me this is looking good. It seems to turn the totally broken batch in MTSJ into something small but working.
> Nice to have is that instead of `@author simon` the JavaDoc should contain some short but reasonable explanation what the code is about. However, not a blocker to me.

Thanks for the hint, I added the missing javadocs.

**Comment by maihacke** [comment]
I also included the example batch for authentication which was on a separate branch before (see #327).

**Comment by hohwille** [comment]
can someone check the travis error?

**Comment by vapadwal** [comment]
> can someone check the travis error?

because of <devon4j.version>3.3.0-SNAPSHOT</devon4j.version> in batch pom

[ERROR] Non-resolvable import POM: Could not find artifact com.devonfw.java.boms:devon4j-bom:pom:3.3.0-SNAPSHOT @ line 50, column 19

**Comment by maihacke** [comment]
Merged PR.
